internal/cmd/group.go

```diff
@@ -144,3 +146,41 @@ func groupsTable(groups []turso.Group) [][]string {
 	}
 	return data
 }
+
+func getGroups(client *turso.Client, fresh ...bool) ([]turso.Group, error) {
+	skipCache := len(fresh) > 0 && fresh[0]
+	if cached := getGroupsCache(client.Org); !skipCache && cached != nil {
+		return cached, nil
+	}
+	groups, err := client.Groups.List()
+	if err != nil {
+		return nil, err
+	}
+	setGroupsCache(client.Org, groups)
+	return groups, nil
+}
+
+func groupNames(client *turso.Client) ([]string, error) {
+	groups, err := getGroups(client)
+	if err != nil {
+		return nil, err
+	}
+	names := make([]string, 0, len(groups))
+	for _, group := range groups {
+		names = append(names, group.Name)
+	}
+	return names, nil
+}
+
+func getGroup(client *turso.Client, name string) (turso.Group, error) {
+	groups, err := getGroups(client)
+	if err != nil {
+		return turso.Group{}, err
+	}
+	for _, group := range groups {
+		if group.Name == name {
+			return group, nil
```

Comment:
just a nit: the happy path being nested made this code a bit more difficult to read. but no need to change this at all